common/middleware/ensure-authenticated.js

```diff
@@ -9,13 +9,33 @@ function _isExpired(authExpiry) {
 module.exports = function ensureAuthenticated({
   provider,
   whitelist = [],
+  expiryMargin = 5 * 60,
 } = {}) {
   return (req, res, next) => {
-    if (whitelist.includes(req.url) || !_isExpired(req.session.authExpiry)) {
+    let authExpiry = req.session.authExpiry
+    if (req.method === 'GET') {
+      authExpiry -= expiryMargin
+    }
+    if (whitelist.includes(req.url) || !_isExpired(authExpiry)) {
       return next()
     }
 
     req.session.originalRequestUrl = req.originalUrl
+    if (req.method === 'POST') {
+      let errorKey
+      const contentType = req.header('content-type')
+      if (contentType.startsWith('multipart/form-data')) {
+        errorKey = 'MULTIPART_FAILED_AUTH'
+      } else if (req.xhr) {
+        errorKey = 'DELETE_FAILED_AUTH'
+      }
+      if (errorKey) {
+        res.status(401)
+        const errorString = req.t(`validation::${errorKey}`)
+        return res.send(errorString)
```

Comment:
Feels like this middleware is now also doing more than its original intention. 

Maybe this block needs to be its own middleware either before or after. But not the middleware to check someone is authenticated is dealing with validation which feels like its crossing concerns quite a bit.

Reply:
Is the issue that I've stored the strings in the validation locales file?

I think this definitely the right place to put these checks. Yes, we could turn the auth expiry check into a module, but this code should only be temporary as per Michael Willis's comment that we should soon have auth tokens that actually last for 12 hours.

Reply:
I think my concern was that this middleware's purpose was quite straightforward before:

- if session not expired or whitelist URL
  - pass to next middleware
- if expired
  - store current URL
  - redirect to auth URL to handle

Now it's doing lots of things crossing over different boundaries:

- if session not expired or whitelist URL
  - pass to next middleware
- if expired
  - if POST request
     - if multipart form
       - kill process and use `res.send` to send error 
     - if XHR request (assume its a file deletion)
       - kill process and use `res.send` to send error 
  - else
    - store current URL
    - redirect to auth URL to handle

Reply:
Updated